test/plugin_functional/test_suites/application_links/redirect_app_links.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import url from 'url';
+import expect from '@kbn/expect';
+import { PluginFunctionalProviderContext } from '../../services';
+import '../../plugins/core_app_status/public/types';
+
+declare global {
+  interface Window {
+    __nonReloadedFlag?: boolean;
+  }
+}
+
+const getKibanaUrl = (pathname?: string, search?: string) =>
+  decodeURIComponent(
+    url.format({
+      protocol: 'http:',
+      hostname: process.env.TEST_KIBANA_HOST || 'localhost',
```

Comment:
Optional: our tests shouldn't rely on the test environment too much since we do not have control over it. We can match against some parts of browser URL instead:
```js
const currentUrl = await browser.getCurrentUrl();
const parsedCurrentUrl = Url.parse(currentUrl);
expect(parsedCurrentUrl.pathname).to.eql('/app/applink_end/some-path');
expect(parsedCurrentUrl.has).to.eql('#/some/hash');
```